lib/nexpose/global_settings.rb

```diff
@@ -115,5 +121,22 @@ def add_control_scanning_to_xml(xml, enabled)
                          'enabled' => enabled ? '1' : '0')
       end
     end
+
+    # Internal method for parsing XML for whether asset linking in enabled.
+    def parse_asset_correlation_from_xml(xml)
+      enabled = false
+      if elem = REXML::XPath.first(xml, '//AssetCorrelation[@enabled]')
+        enabled = elem.attribute('enabled').value.to_i == 1
+      end
+      enabled
+    end
+
+    # Internal method for updating asset linking before saving.
+    def add_asset_correlation_to_xml(xml, enabled)
+      elem = REXML::XPath.first(xml, '//AssetCorrelation')
+      return nil unless elem
```

Comment:
At least in the code for managing scan templates, which I admit is a different beast than global settings, the way it works there is if the corresponding enclosing element is not found, it is added and then the corresponding attributes/elements are added on/below it.  May or may not be a good thing here to do that.